tracex/extraction/prototype/input_handling.py

```diff
@@ -1,215 +1,299 @@
 from . import prompts as p
 
 
-
-def convert_text_to_csv(inp):
+def convert_text_to_csv(text):
     """Converts the input to CSV with intermediate steps."""
     steps = str(7)
     print("Converting Data: Summarizing the text. (1/" + steps + ")", end="\r")
-    bulletpoints = convert_text_to_bulletpoints(inp)
+    dataframe = convert_text_to_bulletpoints(text)
     print(
         "Converting Data: Extracting start date information. (2/" + steps + ")",
         end="\r",
     )
     u.pause_between_queries()
-    bulletpoints_start = add_start_dates(inp, bulletpoints)
+    dataframe = add_start_dates(text, dataframe)
     print(
         "Converting Data: Extracting end date information. (3/" + steps + ")   ",
         end="\r",
     )
     u.pause_between_queries()
-    bulletpoints_end = add_end_dates(inp, bulletpoints_start)
+    dataframe = add_end_dates(text, dataframe)
     print(
         "Converting Data: Extracting duration information. (4/" + steps + ") ", end="\r"
     )
     u.pause_between_queries()
-    bulletpoints_duration = add_durations(inp, bulletpoints_end)
+    dataframe = add_durations(dataframe)
     print(
         "Converting Data: Extracting event types. (5/" + steps + ")          ", end="\r"
     )
     u.pause_between_queries()
-    bulletpoints_event_type = add_event_types(bulletpoints_duration)
+    dataframe = add_event_types(dataframe)
     print(
         "Converting Data: Extracting location information. (6/" + steps + ")", end="\r"
     )
     u.pause_between_queries()
-    bulletpoints_location = add_locations(bulletpoints_event_type)
+    dataframe = add_locations(dataframe)
     print(
         "Converting Data: Creating output CSV. (7/" + steps + ")             ", end="\r"
     )
-    output_path = convert_bulletpoints_to_csv(bulletpoints_location)
+    output_path = convert_dataframe_to_csv(dataframe)
     print("Dataconversion finished.                    ")
+    output_csv(dataframe)
     return output_path
 
 
-def convert_text_to_bulletpoints(inp):
+def convert_text_to_bulletpoints(text):
     """Converts the input text to bulletpoints."""
     messages = [
-        {"role": "system", "content": p.TXT_TO_BULLETPOINTS_CONTEXT},
-        {"role": "user", "content": p.TXT_TO_BULLETPOINTS_PROMPT + inp},
-        {"role": "assistant", "content": p.TXT_TO_BULLETPOINTS_ANSWER},
+        {"role": "system", "content": p.TEXT_TO_EVENTINFORMATION_CONTEXT},
+        {"role": "user", "content": p.TEXT_TO_EVENTINFORMATION_PROMPT + text},
+        {"role": "assistant", "content": p.TEXT_TO_EVENTINFORMATION_ANSWER},
     ]
     bulletpoints = u.query_gpt(messages)
-    bulletpoints = remove_commas(bulletpoints)
-    bulletpoints = add_ending_commas(bulletpoints)
-    with open((u.output_path / "intermediates/1_bulletpoints.txt"), "w") as f:
-        f.write(bulletpoints)
-    return bulletpoints
+    df = pd.DataFrame([], columns=["event_information"])
+    bulletpoints = bulletpoints.replace("- ", "")
```

Comment:
Have you tried to specify in the prompt that the activity labels should be in a specific output format? Then we would not have to do this formatting stuff.